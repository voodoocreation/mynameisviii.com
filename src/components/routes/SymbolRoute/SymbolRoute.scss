@import "../../../scss/variables", "../../../scss/mixins";

.SymbolRoute {
  h2 {
    @include line-before-after();
  }

  section {
    @include panel-1();

    margin-bottom: $spacing;
    padding: $spacing-half $spacing;
    overflow: hidden;
  }

  &--symbol {
    width: 80%;
    margin: auto;

    &::before {
      @include bg-1();
      @include logo-symbol();

      display: block;
      content: "";
      width: 100%;
      padding-top: 100%;
    }
  }

  &--photo {
    @include square-image();
  }

  @include screen-above($screen-xs-min) {
    &--symbol,
    &--photo {
      float: right;
      width: 30%;
      margin-left: $spacing;
      margin-bottom: $spacing;
    }
  }

  @include screen-above(1500px) {
    &--symbol,
    &--photo {
      width: 20%;
    }
  }
}
