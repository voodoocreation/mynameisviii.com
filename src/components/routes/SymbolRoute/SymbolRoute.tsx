import Head from "next/head";
import * as React from "react";
import { FormattedMessage, InjectedIntlProps } from "react-intl";

import { absoluteUrl } from "../../../helpers/dataTransformers";
import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import ButtonBar from "../../presentation/ButtonBar/ButtonBar";
import Image from "../../presentation/Image/Image";
import Link from "../../presentation/Link/Link";
import PageHeader from "../../presentation/PageHeader/PageHeader";

import "./SymbolRoute.scss";

interface IProps extends InjectedIntlProps {}

class SymbolRoute extends React.Component<IProps> {
  public render() {
    const { formatMessage } = this.props.intl;

    const pageTitle = formatMessage({ id: "SYMBOL_TITLE" });
    const pageDescription = formatMessage({ id: "SYMBOL_DESCRIPTION" });

    return (
      <article className="SymbolRoute">
        <Head>
          <title>
            {pageTitle}
            {" Â· "}
            {formatMessage({ id: "BRAND_NAME" })}
          </title>

          <meta content={pageDescription} name="description" />
          <meta property="og:title" content={pageTitle} />
          <meta property="og:description" content={pageDescription} />
          <meta property="og:type" content="website" />
          <meta
            property="og:image"
            content="https://s3.amazonaws.com/mynameisviii-static/theme/heart/og/symbol.jpg"
          />
          <meta property="og:url" content={absoluteUrl("/symbol")} />
        </Head>

        <PageHeader>{pageTitle}</PageHeader>

        <section className="SymbolRoute--meaning">
          <p>
            <FormattedMessage id="SYMBOL_MEANING_CONTENT_1" />
          </p>
          <div className="SymbolRoute--symbol" />
          <p>
            <FormattedMessage id="SYMBOL_MEANING_CONTENT_2" />
          </p>
          <p>
            <FormattedMessage id="SYMBOL_MEANING_CONTENT_3" />
          </p>
          <ul>
            <li>
              <FormattedMessage id="SYMBOL_MEANING_CONTENT_4" />
            </li>
            <li>
              <FormattedMessage id="SYMBOL_MEANING_CONTENT_5" />
            </li>
            <li>
              <FormattedMessage id="SYMBOL_MEANING_CONTENT_6" />
            </li>
          </ul>
          <p>
            <FormattedMessage id="SYMBOL_MEANING_CONTENT_7" />
          </p>
          <p>
            <FormattedMessage id="SYMBOL_MEANING_CONTENT_8" />
          </p>
        </section>

        <section className="SymbolRoute--about">
          <h2>
            <FormattedMessage id="WHO_IS_VIII" />
          </h2>
          <Image
            alt="Viii"
            className="SymbolRoute-photo"
            src="https://s3.amazonaws.com/mynameisviii-static/galleries/dark-spaces-i/09.jpg"
          />
          <p>
            <FormattedMessage id="SYMBOL_ABOUT_CONTENT_1" />
          </p>
          <p>
            <FormattedMessage id="BIO_CONTENT_4" />
          </p>
          <p>
            <FormattedMessage id="SYMBOL_ABOUT_CONTENT_2" />
          </p>
          <ButtonBar>
            <Link className="Button" route="/">
              <FormattedMessage id="EXPLORE_THE_REST_OF_THE_SITE" />
            </Link>
          </ButtonBar>
        </section>
      </article>
    );
  }
}

export default injectIntlIntoPage(SymbolRoute);
