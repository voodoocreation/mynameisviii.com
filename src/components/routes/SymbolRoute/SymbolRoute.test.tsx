import { render } from "enzyme";
import * as React from "react";

import SymbolRoute from "./SymbolRoute";

const setup = (fn: any) => {
  return {
    actual: fn(<SymbolRoute />)
  };
};

describe("[routes] <SymbolRoute />", () => {
  it("renders correctly", () => {
    const { actual } = setup(render);

    expect(actual).toMatchSnapshot();
  });
});
