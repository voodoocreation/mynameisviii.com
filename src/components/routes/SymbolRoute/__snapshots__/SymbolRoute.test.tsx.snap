// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <SymbolRoute /> renders correctly 1`] = `
<article
  class="SymbolRoute"
>
  <header
    class="PageHeader"
  >
    <h1>
      The Viii Symbol
    </h1>
  </header>
  <section
    class="SymbolRoute--meaning"
  >
    <p>
      It appears you have discovered a Viii symbol out in the world – but what does it mean?
    </p>
    <div
      class="SymbolRoute--symbol"
    />
    <p>
      Aside from being an embodiment of the artist known as Viii (pronounced 'Vee'), it also represents the different sides of identity...
    </p>
    <p>
      Each of the points of the triangle represent the start of a field of vision – three different perspectives of an individual:
    </p>
    <ul>
      <li>
        How they see themselves
      </li>
      <li>
        How others see them
      </li>
      <li>
        How they wish to be seen by others
      </li>
    </ul>
    <p>
      The open eye in the centre of the triangle represents the individual themselves, with them being fully self-aware and mindful of these perspectives.
    </p>
    <p>
      The name 'Viii' also correlates to the meanings behind the symbol, with 'V' symbolising the viewpoint/triangle, with three of the letter 'i' representing each 'identity' perspective.
    </p>
  </section>
  <section
    class="SymbolRoute--about"
  >
    <h2>
      Who is Viii?
    </h2>
    <figure
      class="Image SymbolRoute--photo isLoading"
    >
      <div
        class="Image--wrapper"
      >
        <img
          alt="Viii"
          src="https://s3.amazonaws.com/mynameisviii-static/galleries/dark-spaces-i/09.jpg"
        />
        <span
          class="Loader"
        >
          <div
            class="Loader--animation"
          >
            <span />
            <span />
            <span />
          </div>
        </span>
      </div>
    </figure>
    <p>
      Viii is a New Zealand-born singer, songwriter, instramentalist and record producer. This website serves as a means to release the different forms of creative information such as music releases, photoshoots, upcoming shows, design resources, and even song stems for other producers to use to create remixes.
    </p>
    <p>
      Outside of the music, Viii also handles their own artistic direction and other aspects surrounding releases and performances, such as photography, graphic design, software development, makeup artistry and fashion.
    </p>
    <p>
      Consider this discovery as an open invitation to explore the rest of this website and listen to the music that has been released to date.
    </p>
    <div
      class="ButtonBar"
    >
      <a
        class="Button isStyled"
        href="/"
      >
        Explore the rest of the site
      </a>
    </div>
  </section>
</article>
`;
